app/frames/waitlists/[slug]/join/route.tsx

```diff
@@ -55,10 +59,40 @@ const frameHandler = frames(async (ctx) => {
     }
   }
 
-  // In this case you are a referrer and you should be in the waitlist
+  // Get the captcha id, ref and refSquared from the search params
+  const captchaId = ctx.url.searchParams.get("id");
   let ref =
     ctx.url.searchParams.get("ref") || ctx.message.castId?.fid.toString();
+  const refSquared = ctx.url.searchParams.get("refSquared");
+
+  if (waitlist.hasCaptcha && captchaId) {
+    const inputText = ctx.message.inputText;
+    const isCaptchaPassed = await validateCaptchaChallenge(
+      parseInt(captchaId),
+      inputText
+    );
+    if (!isCaptchaPassed) {
```

Comment:
Whether captcha is passed or not, we need to delete that challenge

Reply:
Ok, cool, I did it but then deleted it because I wasn't sure about that ahahah
One question: what happens if a bot clicks join and then stops? The CaptchaChallenge is created but never deleted.
I thought about cleaning up the database everytime a captcha challenge is evaluated, if the challenged expired (it requires to add an expiration date to the CaptchaChallenge model)

Reply:
It's an edge case that I would solve by deleting periodically old captchas, just by looking at the createdAt date